### PR TITLE
Feature/articles pages

### PR DESCRIPTION
**Pull request by eliotmess**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "dependencies": {
     "@babel/plugin-proposal-decorators": "7.7.4",
     "@react-native-community/async-storage": "1.7.1",
+    "@react-native-community/masked-view": "0.1.6",
```
Comment (lukaszblasz) [inline package.json]: why wee need masked view?

File: src/app/pages/ArticleScreen/ArticleScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    this.props.navigation.goBack();
+  };
+
+  private openArticleInBrowser = () => {
+    const { url } = this.props.navigation.state.params?.article;
+    Linking.canOpenURL(url).then(supported => {
+      if (supported) {
+        Linking.openURL(url);
+      } else {
+        throw Error(`Don't know how to open URI: ${url}`);
```
Comment (lukaszblasz) [inline src/app/pages/ArticleScreen/ArticleScreen.tsx]: react intl

File: src/app/pages/ArticleScreen/components/ArticleScreen.styles.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+export const ArticleMetaDataWrapper = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  margin: ${props => props.theme.margin.medium} 0
+    ${props => props.theme.margin.small} 0;
+`;
+
+export const ArticleImage = styled.Image`
+  width: 100%;
+  height: 200;
```
Comment (lukaszblasz) [inline src/app/pages/ArticleScreen/components/ArticleScreen.styles.ts]: no units?

File: src/app/pages/MainArticlesScreen/components/ArticleComponent/index.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import React from 'react';
 import { compose } from 'redux';
 import { injectIntl } from 'react-intl';
+import { NavigationScreenComponent, withNavigation } from 'react-navigation';
 
 import ArticleComponent from './Article.component';
 
-export const Article = compose(injectIntl)(ArticleComponent) as React.ReactType;
+export const Article = withNavigation(
```
Comment (lukaszblasz) [inline src/app/pages/MainArticlesScreen/components/ArticleComponent/index.ts]: something bad happaned here. Compose should wrap all HOCS, witH Natvagation is a HOC and should be inside, 
Also ar eyou able to make correct typings to NavigationScreenComponent? Its its something not trivial to fix any can stay :)

File: src/app/styles/components/buttons/StyledButton.tsx [hunk 5]
```diff
@@ -14,11 +14,12 @@ const StyledButton = ({ onPress, title }: StyledButtonProps): JSX.Element => (
 
 const ButtonContainer = styled.TouchableOpacity`
   width: 100%;
-  height: 35;
+  min-height: 35;
```
Comment (lukaszblasz) [inline src/app/styles/components/buttons/StyledButton.tsx]: px?

File: src/app/pages/MainArticlesScreen/components/ArticleComponent/index.ts (hunk 4 above)
Comment (eliotmess) [reply src/app/pages/MainArticlesScreen/components/ArticleComponent/index.ts]: I tried once before and failed, but I'll try again :)

File: package.json (hunk 1 above)
Comment (eliotmess) [reply package.json]: After upgrading `react-navigation` and `react-navigation-stack` I started getting errors about this dependency.
Similar case to this one: https://github.com/react-navigation/stack/issues/323

File: src/app/pages/ArticleScreen/ArticleScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+class ArticleScreen extends PureComponent<ArticleScreenProps> {
+  private handleBack = () => {
+    this.props.navigation.goBack();
+  };
+
+  private openArticleInBrowser = () => {
+    const { url } = this.props.navigation.state.params?.article;
+    Linking.canOpenURL(url).then(supported => {
+      if (supported) {
+        Linking.openURL(url);
```
[the remark this answers is not in the document]
Comment (eliotmess) [reply src/app/pages/ArticleScreen/ArticleScreen.tsx]: yes, on both ios and android

File: src/app/styles/components/buttons/StyledButton.tsx (hunk 5 above)
Comment (eliotmess) [reply src/app/styles/components/buttons/StyledButton.tsx]: Same as above

File: src/app/pages/ArticleScreen/components/ArticleScreen.styles.ts (hunk 3 above)
Comment (eliotmess) [reply src/app/pages/ArticleScreen/components/ArticleScreen.styles.ts]: I went with default react-native units which are [density-independent pixels](https://facebook.github.io/react-native/docs/height-and-width). Would you like me to change it on pixels?

Comment (lukaszblasz) [reply src/app/pages/ArticleScreen/components/ArticleScreen.styles.ts]: they can stay

**Pull request merged**